aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -85,8 +87,20 @@ Tensor& _clamp_min_out_cpu(Tensor& result, const Tensor& self, Scalar min) {
   return legacy::th::_th_clamp_min_out(result, self, min);
 }
 
-Tensor& fill_(Tensor& self, Scalar value) {
-  return at::legacy::th::_th_fill_(self, value);
+Tensor& _fill__cpu(Tensor& self, Scalar value) {
+  auto iter = TensorIterator::unary_op(self, self);
+  if (iter->dtype() == at::ScalarType::Half) {
```

Comment:
This should be pushed down to `fill_kernel` so it gets auto-vectorized in OSS.

`unary_kernel` will probably get auto-vectorized. If not, use `unary_kernel_vec` with int16_t and `fill_value.x` as the "fill value".